### DISCUSSION TITLE
Optional Single/Multiple UploadFile appears to be text entry instead of Choose File button when using the fastapi docs link

### DISCUSSION THREAD

**Discussion by AbitraryYu** [fastapi/fastapi#11356] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Union

from fastapi import FastAPI, File, UploadFile
from typing_extensions import Annotated

app = FastAPI()


@app.post("/files/")
async def create_file(file: Annotated[Union[bytes, None], File()] = None):
    if not file:
        return {"message": "No file sent"}
    else:
        return {"file_size": len(file)}


@app.post("/uploadfile/")
async def create_upload_file(file: Union[UploadFile, None] = None):
    if not file:
        return {"message": "No upload file sent"}
    else:
        return {"filename": file.filename}
```

### Description

* Use the example in FastAPI -- Optional UploadFile. Python 3.8+
* Run the example code and open the docs
* The input field displays text field instead of choose file field.

![2024-03-27-164109_1917x1027_scrot](https://github.com/tiangolo/fastapi/assets/46883359/ec8bc7e0-1548-4203-a4f1-a735dc7ad373)

I am not sure if this is a bug, because I expect file upload is a "choose file" button

### Operating System

Linux

### FastAPI Version

0.110.0

### Pydantic Version

2.6.2

### Python Version

Python 3.12.1

### Additional Context

I think the expected display is like this:
![2024-03-27-164505_1920x1038_scrot](https://github.com/tiangolo/fastapi/assets/46883359/dc37495d-ad3c-4c99-9f0e-f9c01ecf09ff)

**Comment by YuriiMotov** [comment, marked as answer]
There is a known FastAPI issue here (https://github.com/tiangolo/fastapi/discussions/9709#discussioncomment-6408407). 
To fix problem you should use `SkipJsonSchema[None]` as a workaround:
```
from pydantic.json_schema import SkipJsonSchema

@app.post("/files/")
async def create_file(file: Annotated[Union[bytes, SkipJsonSchema[None]], File()] = None):
    ...
```

**Reply by AbitraryYu** [reply]
Thanks so much 👍
